gst/nnstreamer/elements/gsttensor_generator.c

```diff
@@ -0,0 +1,262 @@
+{
+  /* setup sink pad */
+  self->sinkpad = gst_pad_new_from_static_template (&sink_template, "sink");
+  gst_pad_set_event_function (self->sinkpad,
+      GST_DEBUG_FUNCPTR (gst_tensor_generator_sink_event));
+  gst_element_add_pad (GST_ELEMENT (self), self->sinkpad);
+
+  /* setup src pad */
+  self->srcpad = gst_pad_new_from_static_template (&src_template, "src");
+  gst_pad_set_event_function (self->srcpad,
+      GST_DEBUG_FUNCPTR (gst_tensor_generator_src_event));
+  gst_element_add_pad (GST_ELEMENT (self), self->srcpad);
+  gst_pad_set_chain_function (self->sinkpad,
+      GST_DEBUG_FUNCPTR (gst_tensor_generator_chain));
+}
+
+/**
+ * @brief Function to finalize instance.
+ */
+static void
+gst_tensor_generator_finalize (GObject * object)
+{
+  G_OBJECT_CLASS (parent_class)->finalize (object);
+}
+
+/**
+ * @brief Setter for tensor_generator properties.
+ */
+static void
+gst_tensor_generator_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  UNUSED (value);
+  switch (prop_id) {
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief Getter for tensor_generator properties.
+ */
+static void
+gst_tensor_generator_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec)
+{
+  UNUSED (value);
+  switch (prop_id) {
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief Handle state transition.
+ */
+static GstStateChangeReturn
+gst_tensor_generator_change_state (GstElement * element,
+    GstStateChange transition)
+{
+  GstStateChangeReturn ret;
+
+  switch (transition) {
+    default:
+      break;
+  }
+
+  ret = GST_ELEMENT_CLASS (parent_class)->change_state (element, transition);
+
+  switch (transition) {
+    default:
+      break;
+  }
+
+  return ret;
+}
+
+/**
+ * @brief Handle event on src pad.
+ */
+static gboolean
+gst_tensor_generator_src_event (GstPad * pad, GstObject * parent,
+    GstEvent * event)
+{
+  g_return_val_if_fail (event != NULL, FALSE);
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_SEEK:
+      /* disable seeking */
+      gst_event_unref (event);
+      return FALSE;
+    default:
+      break;
+  }
+
+  return gst_pad_event_default (pad, parent, event);
+}
+
+/**
+ * @brief Handle event on sink pad.
+ */
+static gboolean
+gst_tensor_generator_sink_event (GstPad * pad, GstObject * parent,
+    GstEvent * event)
+{
+  GstTensorGenerator *self;
+  g_return_val_if_fail (event != NULL, FALSE);
+  self = GST_TENSOR_GENERATOR (parent);
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_CAPS:
+    {
+      GstTensorsConfig config;
+      GstStructure *structure;
+      GstCaps *caps;
+
+      gst_event_parse_caps (event, &caps);
+      structure = gst_caps_get_structure (caps, 0);
+      gst_tensors_config_from_structure (&self->in_config, structure);
+      gst_event_unref (event);
```

Comment:
It is needed free in gst_tensor_generator_finalize() after using self->in_config.
And there are some error cases in gst_tensors_config_from_structure(), how about as below?
```
      if (!gst_tensors_config_from_structure (&self->in_config, structure) ||
          !gst_tensors_config_validate (&self->in_config)) {
        gst_tensors_config_free (&self->in_config);
        gst_event_unref (event);
        return FALSE;
      }
```

Reply:
https://github.com/nnstreamer/nnstreamer/pull/4667

Created this PR to remove duplicated code.
I will use `gst_tensors_config_from_cap` after above PR is merged.